### PR TITLE
Updated test_component_availability() to check if commit reference ex…

### PR DESCRIPTION
**Pull request by bgoncalv**
…ists in repo

### CODE REVIEW

File: check_modulemd.py [hunk 1]
```diff
@@ -209,12 +210,25 @@ def test_component_availability(self):
...
-                "Need to check availability of component RPM %s" % p.name)
+            repository = "http://src.fedoraproject.org/cgit/rpms/%s.git" % p.name
+            if p.repository:
+                repository = p.repository
+            ref = "HEAD"
+            if p.ref:
+                ref = p.ref
+            patch_path = "/patch/?id=%s" % ref
+            #Use wget to check if patch exits, --spider just check if page exists, do not download it
+            cmd = "wget " + repository + patch_path + " --spider -O - > /dev/null 2>&1"
```
Comment (igor-raits) [inline check_modulemd.py]: why not to use urllib?

File: check_modulemd.py [hunk 2]
```diff
@@ -209,12 +210,25 @@ def test_component_availability(self):
...
+            try:
+                subprocess.check_output(cmd, shell=True)
+            except subprocess.CalledProcessError as e:
+                self.error("Could not find ref '%s' on '%s'. returned exit status %d; output:\n%s" %
+                           (ref, p.name, e.returncode, e.output))
+
+            print "INFO: Found ref: %s for %s" % (ref, p.name)
```
Comment (igor-raits) [inline check_modulemd.py]: py3-incompatible

File: check_modulemd.py (hunk 1 above)
Comment (bgoncalv) [reply check_modulemd.py]: Can I use urllib to get just HEAD from the url? I do not want to download the whole file.
httplib module might be an option for this.

Comment (mmathesius) [reply check_modulemd.py]: Or the requests module.

File: check_modulemd.py (hunk 2 above)
Comment (mmathesius) [inline check_modulemd.py]: Should use self.log.info() instead of print.

File: check_modulemd.py (hunk 1 above)
Comment (bgoncalv) [reply check_modulemd.py]: yes, I was reading about it right now.

File: check_modulemd.py (hunk 2 above)
Comment (bgoncalv) [reply check_modulemd.py]: indeed.

**Comment by bgoncalv** [comment]
Updated.

File: check_modulemd.py [hunk 3]
```diff
@@ -209,12 +210,24 @@ def test_component_availability(self):
...
+            patch_path = "/patch/?id=%s" % ref
+            url = repository + patch_path
+            resp = requests.head(url)
+            if not resp or resp.status_code != 200:
+                print resp.headers
+                self.error("Could not find ref '%s' on '%s'. returned exit status %d; output:\n%s" %
+                           (ref, p.name, resp.status_code, resp.text))
+            self.log.info("Found ref: %s for %s" % (ref, p.name))
+
         for p in self.mmd.components.modules.values():
```
Comment (mmathesius) [inline check_modulemd.py]: This section should be implemented, too. components.modules isn't used extensively, but at a minimum the [bootstrap module](http://pkgs.fedoraproject.org/cgit/modules/bootstrap.git/plain/bootstrap.yaml) does.

File: check_modulemd.py [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 import os
 import modulemd
+import requests
```
Comment (mmathesius) [inline check_modulemd.py:5]: Be sure to list `python2-requests` as a prerequisite in the README file.

File: check_modulemd.py [hunk 5]
```diff
@@ -209,12 +210,24 @@ def test_component_availability(self):
...
+            repository = "https://src.fedoraproject.org/cgit/rpms/%s.git" % p.name
+            if p.repository:
+                repository = p.repository
+            ref = "HEAD"
+            if p.ref:
+                ref = p.ref
+            patch_path = "/patch/?id=%s" % ref
+            url = repository + patch_path
+            resp = requests.head(url)
+            if not resp or resp.status_code != 200:
```
Comment (igor-raits) [inline check_modulemd.py]: resp.raise_for_status() I think is better

File: check_modulemd.py [hunk 6]
```diff
@@ -209,12 +210,24 @@ def test_component_availability(self):
...
+                print resp.headers
```
Comment (igor-raits) [inline check_modulemd.py]: some debugging stuff I guess

File: check_modulemd.py (hunk 5 above)
Comment (bgoncalv) [reply check_modulemd.py]: To me raise_for_status()  is not clear when it raises exception, it seems to be only if status code are on range of 4XX and 5XX. 3XX range I'd assume we should treat as error.

I can check if status_code is not within 2XX range and report error.

**Pull request merged**